.github/workflows/fix-license-header.py

```diff
@@ -0,0 +1,95 @@
+# Copyright (c) 2016-2021 The Regents of the University of Michigan
+# This file is part of GSD, released under the BSD 2-Clause License.
+
+"""Add license header to files.
+
+This script makes a file header starting with the first line from ``LICENSE``,
+then the lines in ADDITIONAL_HEADER_LINES. For each file given in the arguments,
+it checks if the first lines in the file match the header with a file type
+specific comment prefix added at the beginning of each line. If it does not
+match, it rewrites the initial comments in the file with the given header.
+"""
+
+import argparse
+import sys
+import os
+
+ADDITIONAL_HEADER_LINES = [
+    b'This file is part of GSD, released under the BSD 2-Clause License.'
+]
+
+COMMENT_PREFIX = {
+    '.py': b'# ',
+    '.pxd': b'# ',
+    '.pyx': b'# ',
+    '.c': b'// ',
+    '.h': b'// ',
+    '.rst': b'.. ',
+}
+
+
+def fix_file(f, header_lines, prefix):
+    """Fix one file.
+
+    Return 0 if the file is not modified, 1 if it is.
+    """
+    # Find the initial lines of the file that starts with the comment string
+    line = f.readline()
+    if line.endswith(b'\r\n'):
+        line_ending = b'\r\n'
+    else:
+        line_ending = b'\n'
+
+    # special case, handle shebang
+    shebang = b''
+    if line.startswith(b'#!'):
+        shebang = line
+        line = f.readline()
+
+    file_header = []
+    while line.startswith(prefix):
+        file_header.append(line[len(prefix):].strip())
+        line = f.readline()
+
+    # read the contents of the file
+    file_contents = line + f.read()
+
+    # check if the header is correct
+    if file_header == header_lines:
+        return 0
+    else:
+        # header doesn't match, rewrite file
+        f.seek(0)
+        f.truncate()
+        f.write(shebang)
+        for line in header_lines:
+            f.write(prefix + line + line_ending)
+        f.write(file_contents)
+
+        return 1
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser('Fixes the license headers in files.',)
+    parser.add_argument('filenames', nargs='*', help='Filenames to fix')
+    args = parser.parse_args(sys.argv[1:])
+
+    # build the header
+    header_lines = []
+    with open('LICENSE', 'r+b') as license:
+        header_lines.append(license.readline().strip())
+
+    header_lines.extend(ADDITIONAL_HEADER_LINES)
+
+    return_value = 0
+
+    for filename in args.filenames:
+        with open(filename, 'r+b') as f:
+            extension = os.path.splitext(filename)[1]
+
+            status = fix_file(f=f,
+                              header_lines=header_lines,
+                              prefix=COMMENT_PREFIX[extension])
```

Comment:
Would you want to check if the extension is supported and throw a user-friendly error messasge?

Reply:
Yes, if we keep the comment prefixes in this file. If we make it a command line arg (https://github.com/glotzerlab/gsd/pull/105#issuecomment-846160594) this won't be needed.